lib/OpenAPI/Client.pm

```diff
@@ -37,8 +38,26 @@ has ua => sub { Mojo::UserAgent->new };
 sub call {
   my ($self, $op) = (shift, shift);
   my $code = $self->can($op);
-  return $self->$code(@_) if $code;
-  Carp::croak('[OpenAPI::Client] No such operationId');
+  Carp::croak('[OpenAPI::Client] No such operationId') if !$code;
+  $self->$code(@_);
+}
+
+sub call_p {
+  my ($self, $op) = (shift, shift);
+  my $code = $self->can($op);
+  my $promise = Mojo::Promise->new;
+  if (!$code) {
+    return $promise->reject('[OpenAPI::Client] No such operationId');
```

Comment:
I really like to just `croak` if called with the wrong operationId. I feel it should behave the same as if if you call a missing method on an object.